File: L006/Arrays/firstIdx.cpp
#include<iostream>
#include<vector>
using namespace std;

int firstIdx( vector<int>arr, int target){
        
        int idx = -1;


        for(int i=0; i<arr.size();i++){
            if ( arr[i] == target) {
                idx= i;
                break;
            }
        }


        return idx;
    }   

int main(){

    int n,d;
    cin>>n;

    vector<int>arr(n,0);
    for(int i=0;i<arr.size();i++){
        cin>>arr[i];
    }

    cin>>d;
    cout<<"First index of "<<d<<" is "<<firstIdx(arr,d);
    return 0;
}

File: L006/Arrays/firstIdx.java
import java.util.Scanner;
import java.util.Arrays;

public class firstIdx{

    public static Scanner sc = new Scanner(System.in);

    public static int firstIdx_(int[] arr, int target){
        
        int idx = -1;


        for(int i=0; i<arr.length;i++){
            if ( arr[i] == target) {
                idx= i;
                break;
            }
        }


        return idx;
    }   

    public static void main(String[] args) {

        int n = sc.nextInt();

        int[] arr = new int[n];

        for (int i = 0; i < arr.length; i++) {
            arr[i] = sc.nextInt();
        }
        int d = sc.nextInt();
        System.out.println(firstIdx_(arr, d));

    }
}
